Feat/29/delivery crud (수정)

maengjjin opened the pull request:
# 작업 내용
- 기존 테스트 코드 수정 
-  피드백 기준으로 코드 수정
- 아이템당 배송 생성이 아닌 허브id별로 아이템 묶어서 배송 생성으로 변경

# 리뷰 포인트
1. JPA 양방향 매핑이 올바른 방향인지
DeliveryItem ⟶ Delivery (@ManyToOne)
Delivery ⟶ DeliveryItem (@OneToMany(mappedBy = "delivery"))
해당 매핑 방식이 현 구조에서 적절한지 검토 부탁드립니다.

2. Facade 패턴 도입이 적절한지
배송 생성 흐름이 복잡해져 Facade로 흐름을 묶었는데 패턴 적용이 적절한지 의견 부탁드립니다.

3. 허브 기준 그룹핑 로직이 적절한지
createDeliveriesForOrderItems() 메서드에서 주문 아이템을 허브 ID 기준으로 그룹핑하여 배송을 생성하는 방식이 적합한 설계인지 검토 부탁드립니다

Commit:
test: 배송 생성·조회 테스트 및 스키마 설정 추가

Commit:
feat: 배송 엔티티 생성

Commit:
test: 멱등키 조회 및 생성 테스트

Commit:
feat: 멱등키 엔티티 생성

Commit:
feat: 배송 등록·조회 인터페이스 및 구현체 생성, 배송 상태 추가

Commit:
test: 배송 리포지토리 인터페이스 및 주입 구조 변경에 따른 테스트 코드 수정

Commit:
feat: 멱등키 등록·조회 인터페이스 및 구현체 생성

Commit:
refactor: 비동기 경로 생성 구조 반영을 위해 routePlanId, routesId nullable로 변경

Commit:
test: 구조 변경에 따른 테스트 수정 및 @Import 누락 보완

Commit:
test: 멱등키 도메인 및 서비스 테스트 추가

- 멱등키 도메인 테스트 (상태 검증)
- 멱등키 서비스 테스트 (생성, 조회, 상태변경)
- IdempotencyCommand 객체 추가

Commit:
test: 멱등키 도메인 및 서비스 테스트 추가

- 배송 도메인 테스트 (생성 검증)
- 배송 서비스 테스트 (생성)
- DeliveryCommand,DeliveryResponse 객체 추가

Commit:
feat: base엔티티 추가

Commit:
Merge branch 'refs/heads/develop' into feat/29/delivery-crud

Commit:
refactor: 패키지 구조 정리 및 JPA Auditing 설정 분리

- 패키지 구조 재정리
- @EnableJpaAuditing 어노테이션을 Application에서 분리하여 JpaAuditingConfig로 이동

Commit:
build: Gradle 의존성 설정

Commit:
test: 도메인 모델 및 레파지토리 레이어 테스트 추가

- Delivery, IdempotencyKey 도메인 단위 테스트
- DeliveryRepository, IdempotencyKeyRepository 레파지토리 테스트
- given-when-then 패턴 및 한국어 메서드명 적용

Commit:
feat: 외부 API 클라이언트 인터페이스 및 구현체 추가

- 업체 정보 조회 인터페이스, 배송 담당자 정보 조회 인터페이스 구현체
- 배송담당자, 업체조회 도메인

Commit:
test: 배송 서비스 기본 기능 및 배송 파사드 테스트

- 단일/여러 아이템 배송 생성 테스트
- 멱등키 중복, 외부 API 호출 실패 등 예외 케이스 테스트
- 출발지/도착지 로직 검증

Commit:
feat: 배송 서비스 기본 기능 구현 Facade 패턴을 통한 배송 생성

- 배송 생성, 조회, 상태 변경
- 멱등키 생성 및 상태 관리
- 멱등키 중복, 외부 API 호출 실패 등 예외 케이스 테스트

Commit:
feat: 배송 생성/조회/상태변경 API 컨트롤러 및 DTO 추가

Commit:
test: controller, E2E 통합 테스트 및 테스트 환경 설정

- 테스트 데이터 Fixture 추가

Commit:
doc: tdd readme 생성

Commit:
doc: TEST_GUIDE 생성

leeMK09 commented on delivery/src/main/java/com/klp/delivery/delivery/presentation/dto/DeliveryDetailResponse.java:
```diff
@@ -0,0 +1,32 @@
+    DeliveryStatus status
+) {
+
+  public static DeliveryDetailResponse from(com.klp.delivery.delivery.domain.Delivery delivery) {
```
여기 import 가 패키지 풀 네임까지 import 된 것 같습니다

dnjsals45 replied:
패키지 주소를 상단 import로 변경해도 좋을 것 같아요

p990805 commented on delivery/src/main/java/com/klp/delivery/delivery/presentation/dto/DeliveryDetailResponse.java:
```diff
@@ -0,0 +1,32 @@
+    DeliveryStatus status
+) {
+
+  public static DeliveryDetailResponse from(com.klp.delivery.delivery.domain.Delivery delivery) {
```
```
com.klp.delivery.delivery.domain.
```
이부분은 import로 따로 빼도 괜찮을 거 같습니다.!

maengjjin replied:
수정 완료 했습니다 감사합니다

p990805 reviewed (approved):
고생하셨습니다.!

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/application/facade/DeliveryFacade.java:
```diff
@@ -0,0 +1,99 @@
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@Transactional
```
application 계층에서, 클래스 레벨에서 `@Transactional`을 사용하지 않고, 메소드별로 사용하는 방향으로 컨벤션을 이야기 했었습니다.
이유는, 클래스 레벨에서 적용할 경우, 트랜잭션이 불필요한 메소드의 경우에도 영향을 받게 되기 때문에, 메소드 레벨에서 적용을 하고 휴먼에러로 발생할 수 있는 부분은 코드 리뷰를 통해 검증하는 방향으로 이야기해봤습니다.
<img width="635" height="32" alt="스크린샷 2025-11-10 오후 2 48 34" src="https://github.com/user-attachments/assets/4f3a5d7c-3bb8-47ab-92bb-16521bd2fb48" />

p990805 commented on delivery/src/test/java/com/klp/delivery/delivery/repository/DeliveryRepositoryImplTest.java:
```diff
@@ -0,0 +1,88 @@
+
+
+  @Test
+  void 배송_등록_성공() {
```
배송 등록 할때 필수 요소가 null 인 경우와 그에 따른 실패 테스트케이스도 있으면 좋을 것 같습니다.

p990805 replied:
이 내용은 DeliveryTest에 존재하고 있네요 뒤늦게 확인했습니다.!

leeMK09 commented on delivery/src/main/java/com/klp/delivery/delivery/application/service/IdempotencyKeyService.java:
```diff
@@ -0,0 +1,43 @@
+        .ifPresent(k -> k.updateStatus(IdempotencyStatus.COMPLETED));
+
+
+  }
```
메서드명을 updateIdempotencyStatus 라기보다는 completed 되었다고 하는게 좀더 명확할 것 같아보입니다!
그렇게 되면 해당 메서드의 인자를 command 객체가 아닌 String idempotencyKey 만 있어도 될 것 같아보이네요!

ex) `completed(IdempotencyCommand command)`

maengjjin replied:
추후 COMPLETED 외에도 다양한 상태로 업데이트될 가능성이 있어, update로 범용적으로 처리했습니다.
향후 롤백 처리를 위해 실패 상태 등도 공통적으로 다뤄야 할 것으로 예상돼, 특정 상태(COMPLETED)에 한정하지 않고 update라는 일반 메서드명을 사용했습니다.
혹시 이러한 상황을 더 명확하게 표현할 방법이 있을까요?

leeMK09 replied:
그렇다면 실제 코드상에서 commnad 의 status 를 사용하도록 변경해야할 것 같습니다 

```java
public void updateIdempotencyStatus(IdempotencyCommand command) {
    idempotencyKeyRepository.findByIdempotencyKey(command.idempotencyKey())
        .ifPresent(k -> k.updateStatus(command.status()));
  }
```

Commit:
fix: 매개변수에 포함된 패키지 경로 제거

leeMK09 reviewed (commented):
고생하셨습니다!

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/application/facade/DeliveryFacade.java:
```diff
@@ -0,0 +1,99 @@
+
+    UUID arrivalId = UUID.fromString(company.hubId());
+
+    UUID receiverId = UUID.nameUUIDFromBytes(request.customerId().toString().getBytes());
```
receiverId는 수령 유저의 Id 값인 것 같은데, 유저 Id는 Long으로 설정되어 있는데 별도로 UUID로 변환하신 이유가 있나요?
UUID로 사용할 경우, 수령 유저 추적이 안될 것 같습니다

Commit:
chore: QueryDSL Q파일 Git 추적 제외 설정

p990805 commented on delivery/src/test/java/com/klp/delivery/delivery/domain/IdempotencyKeyTest.java:
```diff
@@ -0,0 +1,25 @@
+public class IdempotencyKeyTest extends MockTest {
+
+  @Test
+  void 멱등키_생성시_멱등키상태는_PENDING_설정_검증() {
```
orderId가 null 일 경우와 멱등키가 "" 이거나 null 일 경우 테스트 케이스도 있으면 좋을 것 같습니다.

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/domain/Delivery.java:
```diff
@@ -0,0 +1,120 @@
+
+  @Comment("수령인 ID")
+  @Column(name = "receiver_id", nullable = false)
+  private UUID receiverId;
```
수령인 ID는 유저 간접참조인데, UUID인 이유가 있을까요?

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/domain/Delivery.java:
```diff
@@ -0,0 +1,120 @@
+
+  @Comment("업체 배송담당자 ID")
+  @Column(name = "vendor_drvier_id", nullable = false)
+  private UUID vendorDrvierId;
```
업체 배송담당자도 마찬가지로, 유저 ID인데 UUID로 저장하면 추적이 안될 것 같습니다

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/exception/DeliveryErrorCode.java:
```diff
@@ -0,0 +1,31 @@
+  @Override
+  public String getMessage() {
+    return message;
+  }
```
저희는 Lombok을 사용하니까
```java
@Getter
@RequiredArgsConstructor
```
로 간단히 작성해도 좋을 것 같습니다

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/presentation/controller/DeliveryController.java:
```diff
@@ -0,0 +1,61 @@
+
+@Slf4j
+@RestController
+@RequestMapping("/api/deliveries")
```
엔드포인트 주소의 경우 컨벤션 상 `/api/{domain}` 방식이 아니라 `/{version}/{domain}` 방식으로 작성하기로 했습니다.
`/v1/deliveries` 로 변경이 필요해보여요
<img width="646" height="154" alt="스크린샷 2025-11-10 오후 2 49 11" src="https://github.com/user-attachments/assets/4fb85058-c35b-44c8-9188-b9b44054e1e9" />

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/presentation/controller/DeliveryController.java:
```diff
@@ -0,0 +1,61 @@
+
+    log.info("배송 생성 성공: orderId={}, deliveryCount={}", request.orderId(),
+        response.deliveries().size());
+    return ResponseEntity.status(HttpStatus.CREATED).body(response);
```
`ResponseEntity.status()` 의 경우 사용하지 않는 것으로 통일하기로 했습니다.
```java
ResponseEntity.created(URI주소).body(response);
```
로 변경되어야 할 것 같습니다
<img width="657" height="144" alt="스크린샷 2025-11-10 오후 2 39 52" src="https://github.com/user-attachments/assets/c42f9454-8a8f-4fa8-8a6b-77ee10929dc1" />

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/presentation/dto/DeliveryCreateRequest.java:
```diff
@@ -0,0 +1,24 @@
+    @NotEmpty(message = "orderItems는 필수이며 최소 1개 이상이어야 합니다.")
+    @Valid
+    List<OrderItemDto> orderItems,
+    @NotNull(message = "idempotencyKey는 필수입니다.")
```
`idempotencyKey`의 경우에는 `@NotNull` 보다는 조금 더 강력한 `@NotBlank` 사용이 더 적절하다고 생각하는데, `"", "   "`이 허용될 경우가 있을까요?

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/presentation/dto/OrderItemDto.java:
```diff
@@ -0,0 +1,19 @@
+    UUID productId,
+    UUID deliveryId,
+    @NotNull(message = "quantity는 필수입니다.")
+    Integer quantity
```
`@Positive` 혹은 `@Min` 등으로 음수가 되지 않게 하거나 0이 아닌 부분을 강제하면 더 좋을 것 같습니다

dnjsals45 reviewed (changes requested):
고생하셨습니다. 테스트 코드 작성 엄청 잘해주셨네요, 리뷰하면서 많이 배우게 됐습니다
```java
@ParameterizedTest
@EnumSource(DeliveryStatus.class)
```
이렇게 하나의 테스트코드로 모든 상태값에 대해 검증할 수 있는 부분도 처음 알게 됐어요!

컨벤션 부분 변경해주시면 좋을 것 같고, receiverId 등에 대해서는 제가 기획적으로 잘못 생각하고 있을 수도 있어서, 그런 것이라면 말씀해주시면 감사하겠습니다!

Commit:
refactor: Delivery 엔티티 필드 수정
- 주문아이템 ID, 발송업체 ID 추가
- 업체 배송담당자 ID 타입을 Long으로 변경

Commit:
refactor: 멱등키 상태 변경 로직 수정
- 기존 특정 상태 전환 로직 제거
- status 값 기반으로 유연한 변경 지원
- 생성 시 동적 상태 설정 가능하도록 개선

Commit:
style: 불필요한 공백 제거

Commit:
refactor: Request와 Command 구조 정리
- OrderItemDto를 DeliveryCreateRequest 내부 클래스로 이동
- DeliveryCommand 필드 추가
- OrderToDeliveryCommand를 request 기반으로 재정의하여 필요한 값만 유지
- Facade와 Service를 새로운 Command 구조에 맞게 수정

Commit:
test: Command 구조 변경에 따른 테스트 코드 수정
- 필드 추가 및 메서드 시그니처 변경 반영
- Fixture 구조에 맞게 리팩터링
- OrderItemFixture 추가 및 DeliveryFixture 정리

Commit:
refactor: 불필요 필드 제거 및 어노테이션 수정
- URL 컨벤션에 맞게 매핑 경로 수정
- Lombok 사용으로 getter 메서드 제거 후 @Getter 추가
- 객체 내 불필요 필드 정리

Commit:
refactor: 불필요 필드 제거 및 어노테이션 수정
- 메서드 레벨로 @Transactional 적용
- 멱등키 동시성 문제 대응을 위해 등록 시 REQUIRES_NEW 트랜잭션 사용

Commit:
style: 필드 순서 정렬
- 가독성 향상을 위해 필드 선언 순서 변경 (idempotencyKey 위치 조정)

Commit:
refactor: Request에서 Command 변환 책임 이관 및 UUID 변환 로직 통합

Commit:
refactor: 배송 상태 정의 수정
- 관련 테스트 코드 수정

Commit:
chore: 불필요한 import 제거

Commit:
test: fixture 변경에 따른 테스트 코드 전체 리팩토링

Commit:
refactor: domain 내 Command 객체를 application.command로 이동하고 전체 패키지 구조 정리

Commit:
feat: 배송 생성 시 허브 단위 묶음 생성 기능 적용
- 같은 허브(hubId)를 가진 아이템들을 하나의 Delivery로 그룹화
- Delivery ↔ DeliveryItem 관계를 OneToMany 매핑으로 설정
- DeliveryItem 엔티티 추가 및 각 orderItem을 DeliveryItem으로 연결
- 기존 orderItem별 개별 배송 생성 로직을 허브별 그룹화 로직으로 변경
- API 응답 및 테스트 코드를 새로운 구조에 맞게 수정

Commit:
Merge branch 'develop' into feat/29/delivery-crud

# Conflicts:
#	delivery/build.gradle
#	delivery/src/test/resources/application-test.yml
#	delivery/src/test/resources/schema.sql

Commit:
refactor: 중복 엔티티 제거 및 패키지 구조 개선
- 중복된 JPAConfig, BaseEntity 삭제
- 역할에 맞게 패키지 구조 재구성

Commit:
chore: 불필요한 중복 의존성 제거 및 Gradle 구성 정리

Commit:
test: 주문 ID 조회 테스트 추가 및 기존 테스트 오류 수정

kmr5326 commented on delivery/src/main/java/com/klp/delivery/delivery/application/service/DeliveryService.java:
@Transactional이 필요하지 않을까요?

kmr5326 commented on delivery/src/main/java/com/klp/delivery/delivery/domain/entity/DeliveryItem.java:
```diff
@@ -0,0 +1,54 @@
+    private UUID deliveryItemId;
+
+    @Comment("배송 ID")
+    @ManyToOne
```
배송아이템은 배송이 필수니까 optional=false 속성 지정되면 좋을 것 같아요

kmr5326 commented on delivery/src/main/java/com/klp/delivery/delivery/presentation/controller/DeliveryController.java:
```diff
@@ -0,0 +1,60 @@
+
+  @GetMapping("/{deliveryId}")
+  public ResponseEntity<DeliveryDetailResponse> getDelivery(@PathVariable UUID deliveryId) {
+    var delivery = deliveryService.getDelivery(deliveryId);
```
var는 팀 컨벤션으로 사용하지 않기로 했었습니당

kmr5326 commented on delivery/src/test/resources/application-test.yml:
```diff
@@ -4,6 +4,10 @@ spring:
+
+  cloud:
+    discovery:
+      enabled: false         #eureka 비활성황
```
서비스 디스커버리 비활성화 하신 이유가 있으실까요??

kmr5326 replied:
아 테스트인거 확인했습니당

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/application/service/DeliveryService.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    public Delivery getDelivery(UUID deliveryId) {
+        return deliveryRepository.findByDeliveryId(deliveryId)
+            .orElseThrow(() -> new BusinessException(DeliveryErrorCode.DELIVERY_NOT_FOUND));
+    }
```
```java
@Repository
@RequiredArgsConstructor
public class DeliveryRepositoryImpl implements DeliveryRepository {
  @Override
  public Optional<Delivery> findByDeliveryId(UUID deliveryId) {
      return deliveryJpaRepository.findByDeliveryId(deliveryId);
  }
}
```
현재 Service 로직에서 Optional로 반환받은 값에 대해서 유효성 체크를 하고 있는데, 이 부분을 Infrastructure 부분에서 하는건 어떨까요?
```java
@Repository
@RequiredArgsConstructor
public class DeliveryRepositoryImpl implements DeliveryRepository {
  @Override
  public Delivery findByDeliveryId(UUID deliveryId) {
      return deliveryRepository.findByDeliveryId(deliveryId)
            .orElseThrow(() -> new BusinessException(DeliveryErrorCode.DELIVERY_NOT_FOUND));
  }
}
```
현재 구조처럼 Application 계층에서 작업하게 된다면, 다른 서비스들에서 getDelivery 메소드가 필요한 경우에 중복 코드가 발생할 수 있을 것 같습니다.

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/application/service/IdempotencyKeyService.java:
```diff
@@ -0,0 +1,44 @@
+    idempotencyKeyRepository.findByIdempotencyKey(command.idempotencyKey())
+        .ifPresent(k -> k.updateStatus(command.status()));
+
+  }
```
더티 체킹을 이용하는 부분 같은데 `@Transactional`이 있어야할 것 같습니다.

dnjsals45 commented on delivery/src/main/java/com/klp/delivery/delivery/presentation/dto/DeliveryCreateRequest.java:
```diff
@@ -0,0 +1,67 @@
+import java.util.UUID;
+
+public record DeliveryCreateRequest(
+    @NotNull(message = "orderId는 필수입니다.")
+    String orderId,
```
요청 DTO에서 UUID로 받지 않고, 변환 과정에서 `UUID.fromString()`을 사용하는 이유가 있을까요? 

그리고 만약 String을 일부러 사용하시는 것이라면, 아래 부분의 idempotencykey 처럼 `@NotBlank`가 적합해보입니다.

dnjsals45 commented on delivery/src/test/java/com/klp/delivery/delivery/application/IdempotencyServiceTest.java:
```diff
@@ -0,0 +1,101 @@
+
+    // when & then: 중복 멱등키로 인한 예외 발생 검증
+    Assertions.assertThrows(
+        com.klp.common.exception.BusinessException.class,
```
패키지명을 상단 import로 올리는 것이 나을 것 같습니다

dnjsals45 commented on delivery/src/test/java/com/klp/delivery/delivery/application/facade/DeliveryFacadeTest.java:
```diff
@@ -0,0 +1,280 @@
+                }
+
+                return delivery;
+            });
```
이 부분에 대한 Mocking을 간소화해서 표현할 수 없을까요? 현재 방식처럼 작성하게 된다면 테스트 코드 내부에 모킹 응답으로 try-catch 구문이 들어가게 되는데 try-catch 구문이 있다는건 에러가 발생할 수 있는 코드라고 생각됩니다.
현재 테스트하고자 하는 부분은 배송생성 성공인데, Exception이 발생할 수 있는 부분이 테스트 코드로 표현되는 것이 어색하다고 느껴집니다.

차라리, Delivery를 2개 만들고, 리플렉션을 이용해서 아이디를 강제로 설정해준다음에, 
`when(deliveryService.registerDelivery()`에서 `.thenReturn(delivery1, devliery2)`와 같은 방식으로 작성하는 것이 단위 테스트라는 목적에 조금 더 부합하지 않을까 생각해봅니다.

dnjsals45 commented on delivery/src/test/java/com/klp/delivery/delivery/presentation/controller/DeliveryIntegrationTest.java:
```diff
@@ -0,0 +1,104 @@
+
+    private static final Logger log = LoggerFactory.getLogger(DeliveryIntegrationTest.class);
+    @LocalServerPort
+  private int port;
```
들여쓰기가 이상하게 되어있습니다. 또한 log는 사용되지 않는 것으로 보입니다.

dnjsals45 reviewed (commented):
고생하셨습니다. 컨벤션적인 부분이나 `@Transactional` 부분에 대해서 확인해주시면 크게 문제는 없어보입니다. 혹시 Indent가 2로 설정되어 있으신 것 같은데... 이 부분은 IDE에서 확인해서 변경해주셨는데도 변경이 안된걸까요..?

dnjsals45 commented:
@maengjjin 
리뷰 포인트를 늦게 확인했네요..!
> JPA 양방향 매핑이 올바른 방향인지 해당 매핑 방식이 현 구조에서 적절한지 검토 부탁드립니다.

어노테이션을 활용한 매핑 방식은 맞는 것 같습니다. 외래키의 주인이 OrderItem쪽에 있는 부분도 적절한 것 같아요.
그런데, 현재 DeliveryItem 생성자를 확인해보면
```java
public static DeliveryItem create(Delivery delivery, UUID orderItemId) {
    return new DeliveryItem(delivery, orderItemId);
}
```
와 같이 작성되어 있습니다. 따라서, Item 생성 시에 Delivery가 할당된 상태이기 때문에 아래에 있는 
```java
public void assignTo(Delivery delivery){
    this.delivery = delivery;
}
```
부분은 불필요한 것 같습니다. 

> Facade 패턴 도입이 적절한지 배송 생성 흐름이 복잡해져 Facade로 흐름을 묶었는데 패턴 적용이 적절한지 의견 부탁드립니다.

Facade 패턴의 도입 자체는 적절한 판단이라고 생각합니다. api 호출 부분에 대한 작업이 이루어졌을 때, 다시 한 번 확인해보고 싶긴 합니다!
Facade가 존재한다면, 굳이 Service 계층에서 호출을 할 필요가 없다고 생각되는 것 같아요


> 허브 기준 그룹핑 로직이 적절한지 createDeliveriesForOrderItems() 메서드에서 주문 아이템을 허브 ID 기준으로 그룹핑하여 배송을 생성하는 방식이 적합한 설계인지 검토 부탁드립니다

저는 그룹핑해서 배송을 생성하는 것이 더 효율적이라서 현재는 적합하다고 판단됩니다.

p990805 commented on delivery/src/main/java/com/klp/delivery/delivery/presentation/dto/DeliveryDetailResponse.java:
```diff
@@ -0,0 +1,33 @@
+    UUID orderId,
+    UUID departureId,
+    UUID arrivalId,
+    UUID receiverId,
```
receiver가 만약 최종 배송 도착지 라면 UUID 가 아닌 Long 일거라고 생각합니다.!

p990805 commented on delivery/src/main/java/com/klp/delivery/delivery/application/command/DeliveryCommand.java:
```diff
@@ -0,0 +1,18 @@
+    UUID departureId,
+    UUID arrivalId,
+    UUID senderId,
+    UUID receiverId,
```
만약 receiver가 최종 목적지 라면 UUID보다 Long 일것 같습니다.!

p990805 reviewed (approved):
고생하셨습니다.!

Commit:
refactor: PR 리뷰 의견 기반 코드 수정

Commit:
style: 코드 정렬
